pkg/api/v1/validation/validation.go

```diff
@@ -70,12 +70,16 @@ func ValidateResourceRequirements(requirements *v1.ResourceRequirements, fldPath
 
 func validateContainerResourceName(value string, fldPath *field.Path) field.ErrorList {
 	allErrs := validateResourceName(value, fldPath)
-	if len(strings.Split(value, "/")) == 1 {
-		if !helper.IsStandardContainerResourceName(value) {
-			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
-		}
+
+	// Container resources must be one of:
+	// - A known resource in the `*kubernetes.io/` namespace.
+	// - An extended resource (not in the `*kubernetes.io/` namespace.
+	// - An opaque integer resource [DEPRECATED] (TODO: Remove following deprecation)
+	if !helper.IsStandardContainerResourceName(value) &&
```

Comment:
Enumerated validation would mean that while upgrading a HA cluster to a version that allowed new kubernetes.io/* resources, skewed masters would reject the newly valid resources, which could lead to a pod being accepted by a new apiserver, but be unable to be updated via an old apiserver. Not sure that's a blocker, but we should at least consider the use case